vita3k/modules/SceLibKernel/SceLibKernel.cpp

```diff
@@ -1557,8 +1557,24 @@ EXPORT(int, sceKernelWaitSemaCB, SceUID semaid, int signal, SceUInt *timeout) {
     return semaphore_wait(host.kernel, export_name, thread_id, semaid, signal, timeout);
 }
 
-EXPORT(int, sceKernelWaitSignal) {
-    return UNIMPLEMENTED();
+struct SceKernelSignalWaitParams {
+    uint32_t reserved[2];
+    // TODO figure out stack_addr (maybe struct?)
+    Address stack_addr;
+};
+
+EXPORT(int, sceKernelWaitSignal, uint32_t unknown, uint32_t delay, uint32_t timeout, Address params_addr) {
```

Comment:
You can make `params_addr` a `SceKernelSignalWaitParams *` and it will work! Give it a shot.

Reply:
cool! it works